### PR TITLE
OCM-9035 | ci: Automated 74096

### PR DESCRIPTION
**Pull request by xueli181114**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, use `fixes #<issue_number>(, fixes #<issue_number>, ...)` format, where issue_number might be a GitHub issue, or a Jira story (OCM-xxxx)*:
Fixes #
[OCM-9035](https://issues.redhat.com//browse/OCM-9035)
**Change type**
- [ ] New feature
- [ ] Bug fix
- [ ] Build
- [x] CI
- [ ] Documentation
- [ ] Performance
- [ ] Refactor
- [ ] Style
- [ ] Unit tests
- [ ] Subsystem tests

**Checklist**
- [ ] Subject and description added to both, commit and PR.
- [ ] Relevant issues have been referenced.

### CODE REVIEW

**Comment by xueli181114** [comment]
```
lixue@Xue-Lis-MacBook-Pro terraform-provider-rhcs % ginkgo -label-filter '(Critical,High)&&day1-post&&!Exclude' tests/e2e
time="2024-07-17T18:24:42+08:00" level=info msg="Running against RHCS URL: https://api.stage.openshift.com"
Running Suite: e2e tests suite - /Users/lixue/Workspace/ocm-tf/terraform-provider-rhcs/tests/e2e
================================================================================================
Random Seed: 1721211875

Will run 11 of 113 specs
time="2024-07-17T18:24:44+08:00" level=info msg="Loaded cluster profile configuration from profile rosa-hcp-full : map[byovpc:true channel_group:stable cloud_provider:aws cluster_type:rosa-hcp compute_machine_type:m5.xlarge deactive_cluster_waiter:true full_resources:true major_version:4.15 multi_az:false oidc_config:un-managed product_id:rosa region:us-west-2 sts:true]"
time="2024-07-17T18:24:44+08:00" level=info msg="Running terraform init against the dir /Users/lixue/Workspace/ocm-tf/terraform-provider-rhcs/tests/tf-manifests/rhcs/clusters/rosa-hcp"
time="2024-07-17T18:24:44+08:00" level=debug msg="Running terraform command: [init -no-color]"
time="2024-07-17T18:24:49+08:00" level=debug msg="Initializing the backend...\n\nInitializing provider plugins...\n- Reusing previous version of hashicorp/aws from the dependency lock file\n- Reusing previous version of terraform.local/local/rhcs from the dependency lock file\n- Reusing previous version of hashicorp/random from the dependency lock file\n- Using previously-installed hashicorp/aws v5.58.0\n- Using previously-installed terraform.local/local/rhcs v1.6.0\n- Using previously-installed hashicorp/random v3.6.2\n\nTerraform has been successfully initialized!\n\nYou may now begin working with Terraform. Try running \"terraform plan\" to see\nany changes that are required for your infrastructure. All Terraform commands\nshould now work.\n\nIf you ever set or change modules or backend configuration for Terraform,\nrerun this command to reinitialize your working directory. If you forget, other\ncommands will detect it and remind you to do so if necessary."
time="2024-07-17T18:24:52+08:00" level=debug msg="{\n  \"cluster_id\": \"2cik1uhcrskf3jbmoa2mn2ne5s1q8dnr\",\n  \"cluster_name\": \"rhcs-hcp-full-46s\",\n  \"cluster_version\": \"4.15.20\",\n  \"properties\": {\n    \"custom_property\": \"test\",\n    \"qe_usage\": \"\",\n    \"rosa_creator_arn\": \"arn:aws:iam::301721915996:user/xueli\"\n  }\n}"
SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSStime="2024-07-17T18:24:52+08:00" level=info msg="Loaded cluster profile configuration from profile rosa-hcp-full : map[byovpc:true channel_group:stable cloud_provider:aws cluster_type:rosa-hcp compute_machine_type:m5.xlarge deactive_cluster_waiter:true full_resources:true major_version:4.15 multi_az:false oidc_config:un-managed product_id:rosa region:us-west-2 sts:true]"
Stime="2024-07-17T18:24:53+08:00" level=info msg="Loaded cluster profile configuration from profile rosa-hcp-full : map[byovpc:true channel_group:stable cloud_provider:aws cluster_type:rosa-hcp compute_machine_type:m5.xlarge deactive_cluster_waiter:true full_resources:true major_version:4.15 multi_az:false oidc_config:un-managed product_id:rosa region:us-west-2 sts:true]"
•Stime="2024-07-17T18:24:54+08:00" level=info msg="Loaded cluster profile configuration from profile rosa-hcp-full : map[byovpc:true channel_group:stable cloud_provider:aws cluster_type:rosa-hcp compute_machine_type:m5.xlarge deactive_cluster_waiter:true full_resources:true major_version:4.15 multi_az:false oidc_config:un-managed product_id:rosa region:us-west-2 sts:true]"
•time="2024-07-17T18:24:55+08:00" level=info msg="Loaded cluster profile configuration from profile rosa-hcp-full : map[byovpc:true channel_group:stable cloud_provider:aws cluster_type:rosa-hcp compute_machine_type:m5.xlarge deactive_cluster_waiter:true full_resources:true major_version:4.15 multi_az:false oidc_config:un-managed product_id:rosa region:us-west-2 sts:true]"
•time="2024-07-17T18:24:55+08:00" level=info msg="Loaded cluster profile configuration from profile rosa-hcp-full : map[byovpc:true channel_group:stable cloud_provider:aws cluster_type:rosa-hcp compute_machine_type:m5.xlarge deactive_cluster_waiter:true full_resources:true major_version:4.15 multi_az:false oidc_config:un-managed product_id:rosa region:us-west-2 sts:true]"
•SSStime="2024-07-17T18:24:56+08:00" level=info msg="Loaded cluster profile configuration from profile rosa-hcp-full : map[byovpc:true channel_group:stable cloud_provider:aws cluster_type:rosa-hcp compute_machine_type:m5.xlarge deactive_cluster_waiter:true full_resources:true major_version:4.15 multi_az:false oidc_config:un-managed product_id:rosa region:us-west-2 sts:true]"
•time="2024-07-17T18:24:56+08:00" level=info msg="Loaded cluster profile configuration from profile rosa-hcp-full : map[byovpc:true channel_group:stable cloud_provider:aws cluster_type:rosa-hcp compute_machine_type:m5.xlarge deactive_cluster_waiter:true full_resources:true major_version:4.15 multi_az:false oidc_config:un-managed product_id:rosa region:us-west-2 sts:true]"
•SStime="2024-07-17T18:24:57+08:00" level=info msg="Loaded cluster profile configuration from profile rosa-hcp-full : map[byovpc:true channel_group:stable cloud_provider:aws cluster_type:rosa-hcp compute_machine_type:m5.xlarge deactive_cluster_waiter:true full_resources:true major_version:4.15 multi_az:false oidc_config:un-managed product_id:rosa region:us-west-2 sts:true]"
Stime="2024-07-17T18:24:58+08:00" level=info msg="Loaded cluster profile configuration from profile rosa-hcp-full : map[byovpc:true channel_group:stable cloud_provider:aws cluster_type:rosa-hcp compute_machine_type:m5.xlarge deactive_cluster_waiter:true full_resources:true major_version:4.15 multi_az:false oidc_config:un-managed product_id:rosa region:us-west-2 sts:true]"
SSStime="2024-07-17T18:24:58+08:00" level=info msg="Loaded cluster profile configuration from profile rosa-hcp-full : map[byovpc:true channel_group:stable cloud_provider:aws cluster_type:rosa-hcp compute_machine_type:m5.xlarge deactive_cluster_waiter:true full_resources:true major_version:4.15 multi_az:false oidc_config:un-managed product_id:rosa region:us-west-2 sts:true]"
Stime="2024-07-17T18:24:58+08:00" level=info msg="Loaded cluster profile configuration from profile rosa-hcp-full : map[byovpc:true channel_group:stable cloud_provider:aws cluster_type:rosa-hcp compute_machine_type:m5.xlarge deactive_cluster_waiter:true full_resources:true major_version:4.15 multi_az:false oidc_config:un-managed product_id:rosa region:us-west-2 sts:true]"
•SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSS

Ran 7 of 113 Specs in 17.683 seconds
SUCCESS! -- 7 Passed | 0 Failed | 0 Pending | 106 Skipped
PASS

Ginkgo ran 1 suite in 26.325227175s
Test Suite Passed
```

File: log
Comment (radtriste) [inline log:1]: I think that file can be removed :)

File: tests/e2e/verification_post_day1_test.go [hunk 1]
```diff
@@ -324,4 +324,76 @@ var _ = Describe("Verify cluster", func() {
 			Expect(cluster.AWS().Ec2MetadataHttpTokens()).To(Equal(cmsv1.Ec2MetadataHttpTokensOptional))
 		}
 	})
+
+	It("reourses will wait for cluster ready - [id:74096]", ci.Day1Post, ci.Critical,
```
Comment (radtriste) [inline tests/e2e/verification_post_day1_test.go:328]:
```suggestion
	It("resourses will wait for cluster ready - [id:74096]", ci.Day1Post, ci.Critical,
```

File: tests/utils/constants/manifests.go [hunk 2]
```diff
@@ -81,6 +81,7 @@ func GetAWSVPCDefaultManifestDir(clusterType ClusterType) string {
 // Dirs of rhcs provider
 var (
 	ClusterDir                  = path.Join(ManifestsConfigurationDir, RHCSProviderDir, "clusters")
+	FullResourcesDir            = path.Join(ManifestsConfigurationDir, RHCSProviderDir, "clusters", "full-resources")
```
Comment (radtriste) [inline tests/utils/constants/manifests.go:84]: is that used ?

File: tests/tf-manifests/rhcs/clusters/rosa-hcp/main.tf
Comment (radtriste) [inline tests/tf-manifests/rhcs/clusters/rosa-hcp/main.tf:1]: We could also define an ingress for HCP with `listening_method="internal"`. and then the listening method into the test case. wdyt ?

File: tests/e2e/verification_post_day1_test.go (hunk 1 above)
Comment (xueli181114) [reply tests/e2e/verification_post_day1_test.go]: Fixed

File: log
Comment (xueli181114) [reply log:1]: Removed

File: tests/utils/constants/manifests.go (hunk 2 above)
Comment (xueli181114) [reply tests/utils/constants/manifests.go]: Removed

File: tests/tf-manifests/rhcs/clusters/rosa-hcp/main.tf
Comment (xueli181114) [reply tests/tf-manifests/rhcs/clusters/rosa-hcp/main.tf:1]: Outof date now.

**Comment by xueli181114** [comment]
```
lixue@Xue-Lis-MacBook-Pro terraform-provider-rhcs % ginkgo -label-filter '(Critical,High)&&day1-post&&!Exclude' tests/e2e
time="2024-07-19T18:42:35+08:00" level=info msg="Running against RHCS URL: https://api.stage.openshift.com"
Running Suite: e2e tests suite - /Users/lixue/Workspace/ocm-tf/terraform-provider-rhcs/tests/e2e
================================================================================================
Random Seed: 1721385745

Will run 12 of 118 specs
time="2024-07-19T18:42:36+08:00" level=info msg="Loaded cluster profile configuration from profile rosa-classic-full : map[byovpc:true channel_group:stable cloud_provider:aws cluster_type:rosa-classic compute_machine_type:m5.xlarge full_resources:true major_version:4.15 multi_az:false no_wait_cluster:true oidc_config:managed product_id:rosa region:ap-northeast-1 sts:true]"
time="2024-07-19T18:42:36+08:00" level=info msg="Running terraform init against the dir /Users/lixue/Workspace/ocm-tf/terraform-provider-rhcs/tests/tf-manifests/rhcs/clusters/rosa-classic"
time="2024-07-19T18:42:36+08:00" level=debug msg="Running terraform command: [init -no-color]"
time="2024-07-19T18:42:42+08:00" level=debug msg="Initializing the backend...\n\nInitializing provider plugins...\n- Reusing previous version of hashicorp/aws from the dependency lock file\n- Reusing previous version of terraform.local/local/rhcs from the dependency lock file\n- Reusing previous version of hashicorp/random from the dependency lock file\n- Using previously-installed hashicorp/random v3.6.2\n- Using previously-installed hashicorp/aws v5.58.0\n- Using previously-installed terraform.local/local/rhcs v1.6.0\n\nTerraform has been successfully initialized!\n\nYou may now begin working with Terraform. Try running \"terraform plan\" to see\nany changes that are required for your infrastructure. All Terraform commands\nshould now work.\n\nIf you ever set or change modules or backend configuration for Terraform,\nrerun this command to reinitialize your working directory. If you forget, other\ncommands will detect it and remind you to do so if necessary."
time="2024-07-19T18:42:45+08:00" level=debug msg="{\n  \"cluster_id\": \"2cju1r5j821md20u14bkr7ha4q4qg1hf\",\n  \"cluster_name\": \"rhcs-classic-full-8gx\",\n  \"cluster_version\": \"4.15.21\"\n}"
SSSSSSSSSSSSSSSStime="2024-07-19T18:42:45+08:00" level=info msg="Loaded cluster profile configuration from profile rosa-classic-full : map[byovpc:true channel_group:stable cloud_provider:aws cluster_type:rosa-classic compute_machine_type:m5.xlarge full_resources:true major_version:4.15 multi_az:false no_wait_cluster:true oidc_config:managed product_id:rosa region:ap-northeast-1 sts:true]"
Stime="2024-07-19T18:42:47+08:00" level=info msg="Loaded cluster profile configuration from profile rosa-classic-full : map[byovpc:true channel_group:stable cloud_provider:aws cluster_type:rosa-classic compute_machine_type:m5.xlarge full_resources:true major_version:4.15 multi_az:false no_wait_cluster:true oidc_config:managed product_id:rosa region:ap-northeast-1 sts:true]"
•Stime="2024-07-19T18:42:48+08:00" level=info msg="Loaded cluster profile configuration from profile rosa-classic-full : map[byovpc:true channel_group:stable cloud_provider:aws cluster_type:rosa-classic compute_machine_type:m5.xlarge full_resources:true major_version:4.15 multi_az:false no_wait_cluster:true oidc_config:managed product_id:rosa region:ap-northeast-1 sts:true]"
•time="2024-07-19T18:42:48+08:00" level=info msg="Loaded cluster profile configuration from profile rosa-classic-full : map[byovpc:true channel_group:stable cloud_provider:aws cluster_type:rosa-classic compute_machine_type:m5.xlarge full_resources:true major_version:4.15 multi_az:false no_wait_cluster:true oidc_config:managed product_id:rosa region:ap-northeast-1 sts:true]"
•time="2024-07-19T18:42:48+08:00" level=info msg="Loaded cluster profile configuration from profile rosa-classic-full : map[byovpc:true channel_group:stable cloud_provider:aws cluster_type:rosa-classic compute_machine_type:m5.xlarge full_resources:true major_version:4.15 multi_az:false no_wait_cluster:true oidc_config:managed product_id:rosa region:ap-northeast-1 sts:true]"
•time="2024-07-19T18:42:49+08:00" level=info msg="Loaded cluster profile configuration from profile rosa-classic-full : map[byovpc:true channel_group:stable cloud_provider:aws cluster_type:rosa-classic compute_machine_type:m5.xlarge full_resources:true major_version:4.15 multi_az:false no_wait_cluster:true oidc_config:managed product_id:rosa region:ap-northeast-1 sts:true]"
SSSStime="2024-07-19T18:42:49+08:00" level=info msg="Loaded cluster profile configuration from profile rosa-classic-full : map[byovpc:true channel_group:stable cloud_provider:aws cluster_type:rosa-classic compute_machine_type:m5.xlarge full_resources:true major_version:4.15 multi_az:false no_wait_cluster:true oidc_config:managed product_id:rosa region:ap-northeast-1 sts:true]"
•time="2024-07-19T18:42:50+08:00" level=info msg="Loaded cluster profile configuration from profile rosa-classic-full : map[byovpc:true channel_group:stable cloud_provider:aws cluster_type:rosa-classic compute_machine_type:m5.xlarge full_resources:true major_version:4.15 multi_az:false no_wait_cluster:true oidc_config:managed product_id:rosa region:ap-northeast-1 sts:true]"
•SStime="2024-07-19T18:42:50+08:00" level=info msg="Loaded cluster profile configuration from profile rosa-classic-full : map[byovpc:true channel_group:stable cloud_provider:aws cluster_type:rosa-classic compute_machine_type:m5.xlarge full_resources:true major_version:4.15 multi_az:false no_wait_cluster:true oidc_config:managed product_id:rosa region:ap-northeast-1 sts:true]"
•time="2024-07-19T18:42:51+08:00" level=info msg="Loaded cluster profile configuration from profile rosa-classic-full : map[byovpc:true channel_group:stable cloud_provider:aws cluster_type:rosa-classic compute_machine_type:m5.xlarge full_resources:true major_version:4.15 multi_az:false no_wait_cluster:true oidc_config:managed product_id:rosa region:ap-northeast-1 sts:true]"
•SStime="2024-07-19T18:42:51+08:00" level=info msg="Loaded cluster profile configuration from profile rosa-classic-full : map[byovpc:true channel_group:stable cloud_provider:aws cluster_type:rosa-classic compute_machine_type:m5.xlarge full_resources:true major_version:4.15 multi_az:false no_wait_cluster:true oidc_config:managed product_id:rosa region:ap-northeast-1 sts:true]"
•SStime="2024-07-19T18:42:52+08:00" level=info msg="Loaded cluster profile configuration from profile rosa-classic-full : map[byovpc:true channel_group:stable cloud_provider:aws cluster_type:rosa-classic compute_machine_type:m5.xlarge full_resources:true major_version:4.15 multi_az:false no_wait_cluster:true oidc_config:managed product_id:rosa region:ap-northeast-1 sts:true]"
•SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSS

Ran 10 of 118 Specs in 20.061 seconds
SUCCESS! -- 10 Passed | 0 Failed | 0 Pending | 108 Skipped
PASS

Ginkgo ran 1 suite in 30.789274788s
Test Suite Passed
```

File: tests/tf-manifests/rhcs/clusters/rosa-hcp/main.tf [hunk 3]
```diff
@@ -93,3 +93,78 @@ resource "rhcs_hcp_default_ingress" "current" {
   cluster          = rhcs_cluster_rosa_hcp.rosa_hcp_cluster.id
```
Comment (radtriste) [inline tests/tf-manifests/rhcs/clusters/rosa-hcp/main.tf:93]: shouldn't we update the `rhcs_hcp_default_ingress` with `var.full_resources` check in `count ` ?

File: tests/e2e/verification_post_day1_test.go [hunk 4]
```diff
@@ -386,4 +386,75 @@ var _ = Describe("Verify cluster", func() {
...
+				Expect(kubeConfig.PodPidsLimit()).To(Equal(4097))
+
+			}
+
+			By("Check the default ingress")
+			if !cluster.Hypershift().Enabled() {
+				ingress, err := cms.RetrieveClusterIngress(ci.RHCSConnection, clusterID)
+				Expect(err).ToNot(HaveOccurred())
+				Expect(ingress.ExcludedNamespaces()).To(ContainElement("full-resource"))
+			}
```
Comment (radtriste) [inline tests/e2e/verification_post_day1_test.go:429]: And then check the default ingress listening method here ?

**Review by radtriste — APPROVED** [review]
minor comments

File: tests/utils/exec/cluster.go [hunk 5]
```diff
@@ -55,6 +55,8 @@ type ClusterArgs struct {
 	IngressListeningMethod *string `hcl:"ingress_listening_method"`
```
Comment (radtriste) [inline tests/utils/exec/cluster.go:55]: Please remove that one

File: tests/ci/profiles/tf_hcp_cluster_profiles.yml [hunk 6]
```diff
@@ -208,4 +209,17 @@ profiles:
     oidc_config: "managed" # id:72493
     admin_enabled: false # Not yet supported
     unified_acc_role_path: ""
-    ingress_listening_method: "internal" # id:72516
\ No newline at end of file
+
```
Comment (radtriste) [inline tests/ci/profiles/tf_hcp_cluster_profiles.yml:212]: I think we can remove the `rosa-hcp-ingress` profile

**Review by radtriste — APPROVED** [review]
/lgtm

**Pull request merged**